pallets/settlement/src/lib.rs

```diff
@@ -1133,35 +1193,27 @@ decl_module! {
         ///
         /// # Errors
         /// * `InstructionNotFailed` - Instruction not in a failed state or does not exist.
-        #[weight = <T as Config>::WeightInfo::execute_manual_instruction(*legs_count)]
-        pub fn execute_manual_instruction(origin, id: InstructionId, legs_count: u32, portfolio: Option<PortfolioId>) {
-            // check origin has the permissions required and valid instruction
-            let (did, sk, instruction_details) = Self::ensure_origin_perm_and_instruction_validity(origin, id, true)?;
-
-            // Check for portfolio
-            let instruction_legs: Vec<(LegId, LegV2)> = Self::get_instruction_legs(&id);
-            match portfolio {
-                Some(portfolio) => {
-                    // Ensure that the caller is a party of this instruction.
-                    T::Portfolio::ensure_portfolio_custody_and_permission(portfolio, did, sk.as_ref())?;
-                    ensure!(
-                        instruction_legs.iter().any(|(_, leg)| leg.from == portfolio || leg.to == portfolio),
-                        Error::<T>::CallerIsNotAParty
-                    );
-                }
-                None => {
-                    // Ensure venue exists & sender is its creator.
-                    Self::venue_for_management(instruction_details.venue_id, did)?;
-                }
-            }
-
-            // check that the instruction leg count matches
-            ensure!(instruction_legs.len() as u32 <= legs_count, Error::<T>::LegCountTooSmall);
-
-            // Executes the instruction
-            Self::execute_instruction_retryable(id)?;
-
-            Self::deposit_event(RawEvent::SettlementManuallyExecuted(did, id));
+        #[weight =
+            <T as Config>::WeightInfo::execute_manual_instruction_weight(weight_limit, legs_count)
+            .max(<T as Config>::WeightInfo::execute_manual_instruction(0))
```

Comment:
We might be able to call `Self::ensure_manual_weight_limit()` here.  If `ensure_manual_weight_limit` always returns at least the minimum weight (instead of throwing an error), then it can be used for both the weight and to initialize the meter.